chirho/counterfactual/ops.py

```diff
@@ -23,10 +25,61 @@ def split(obs: T, acts: Tuple[Intervention[T], ...], **kwargs) -> T:
     return scatter(act_values, event_dim=kwargs.get("event_dim", 0))
 
 
+@pyro.poutine.runtime.effectful(type="undo_split")
+def undo_split(
+    antecedents: Iterable[str] = None, event_dim: int = 0
+) -> Callable[[T], T]:
+    """
+    A helper function that undoes an upstream `chirho.counterfactual.ops.split`
+    operation by gathering the factual value and scattering it back into
+    two alternative cases.
+
+    :param antecedents: A list of upstream intervened sites which induced
+                        the `split` to be reversed.
+    :param event_dim: The event dimension.
```

Comment:
nit: clarify whose event dimension this is (the value to be preempted)

Reply:
done